### DISCUSSION TITLE
How to implement automatic parking strategy when the idle time exceeds a certain threshold.

### DISCUSSION THREAD

**Discussion by jrt324** [openTCS/opentcs#95] [Q&A] [answered]
How to implement such a strategy: When the vehicle is in the Idle state for more than for example 10 minutes after completing a task, it automatically returns to a specified position (which could be a recharging location or the parking position)

I found this in the User's Guide , 
But the document does not provide detailed instructions on how to set up the automatic parking strategy when the idle time exceeds a certain threshold.

**Follow is User's Guide 6.6:**

> ### 6.6. Configuring automatic parking and recharging
> By default, idle vehicles remain where they are after processing their last transport order. You can change this in the kernel’s configuration file:

> - To order vehicles to charging locations automatically, set the configuration parameter defaultdispatcher.rechargeIdleVehicles to true. The default dispatcher will then look for locations at which the idle vehicle’s recharge operation is possible and create orders to send it to such a location (if unoccupied). (Note that the string used for the operation is driver-specific.)

> - To order vehicles to parking positions automatically, set the configuration parameter defaultdispatcher.parkIdleVehicles to true. The default dispatcher will then look for unoccupied parking positions and create orders to send the idle vehicle there.

**Comment by swltr** [comment, marked as answer]
@jrt324 I can see this being useful with some use cases, but the default dispatcher currently does not do this. When completing a transport order and not getting a new one, vehicles are either left where they are or immediately sent to a parking position or recharge location.

You would have to register your own dispatcher implementation for this. Alternatively, you may be able to work around this by using order sequences whose _complete_ flag you set only if either there is a new transportation need for the same vehicle or the waiting time has run out.
